### PR TITLE
Fix: Client not selected when generating invoice from project details page

### PR DESCRIPTION
**Pull request by mayankagnihotri7**
## Notion card
[Notion Card](https://www.notion.so/saeloun/Client-not-getting-selected-automatically-when-generating-an-invoice-from-the-project-details-page-95a331a293ed4031946ce5bcbb09d49b)

## Summary

Project names that were more than one word like `Dell Technologies` was appended to URL as query params and was only getting split using `?` operator whereas for multiple word names there were special characters being inserted by the browser and because of that the name was not matching when filtering to get the data.

## Preview

Video - https://www.awesomescreenshot.com/video/10966971

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [ ] I have added automated tests for my code

### CODE REVIEW

**Review by apoorv1316 — APPROVED** [review]
@mayankagnihotri7 Please rebase

**Comment by mayankagnihotri7** [comment]
@apoorv1316 I have rebased it with the develop.

**Review by JijoBose — APPROVED** [review]

File: app/javascript/src/components/Invoices/Generate/ClientSelection.tsx [hunk 1]
```diff
@@ -31,10 +31,13 @@ const ClientSelection = ({ clientList, selectedClient, setSelectedClient, option
...
 
   React.useEffect(() => {
-    const prePopulatedClient = window.location.search.split("?")[1];
+    const prePopulatedClient = window.location.search.replace(/[^A-Za-z]+/g, "");
+
     if (prePopulatedClient) {
       const selection = clientList.filter(
-        (client) => client.label == prePopulatedClient
+        (client) =>
+          client.label.split(" ").join("").split(",").join("") ==
```
Comment (keshavbiswa) [inline app/javascript/src/components/Invoices/Generate/ClientSelection.tsx:39]:
```suggestion
          client.label.replace(/\s|,/g, '')
```
Can you check if this works?

Comment (mayankagnihotri7) [reply app/javascript/src/components/Invoices/Generate/ClientSelection.tsx]: It is working.

**Review by keshavbiswa — APPROVED** [review]
LGTM!

**Review by apoorv1316 — CHANGES_REQUESTED** [review]
@mayankagnihotri7 This is not working if a client name has a "." or "_" or "-".  It will not work if a client name is "cf.",  "coca-cola" or "coca_cola". Also, see the loom for more scenarios https://www.loom.com/share/acc5c15ceb124608b97690848142f3c8

**Comment by mayankagnihotri7** [comment]
@apoorv1316 I have made the requested changes and handled all those scenarios. Please review.

File: app/javascript/src/components/Invoices/Generate/ClientSelection.tsx [hunk 2]
```diff
@@ -31,11 +31,11 @@ const ClientSelection = ({ clientList, selectedClient, setSelectedClient, option
...
 
   React.useEffect(() => {
-    const prePopulatedClient = window.location.search.replace(/[^A-Za-z]+/g, "");
+    const prePopulatedClient = window.location.search.split("?").join("").replace(/%20/g, " ");
 
     if (prePopulatedClient) {
       const selection = clientList.filter(
-        (client) => client.label.replace(/\s|,/g, "") == prePopulatedClient
+        (client) => client.label == prePopulatedClient
       );
```
Comment (keshavbiswa) [inline app/javascript/src/components/Invoices/Generate/ClientSelection.tsx:39]: Could you explain what are we trying to do with this bit of code?

Comment (mayankagnihotri7) [reply app/javascript/src/components/Invoices/Generate/ClientSelection.tsx:39]: I am replacing spaces which is encoded into the query string by the browser in the form of `%20` and replacing it with the space itself. And for the line `#38` I just reverted the code back to as it was before i.e; comparing the label coming from the backend with the query string we just got after removing the spaces from the url.

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**